router/usecase/router_usecase.go

```diff
@@ -448,6 +439,50 @@ func (r *routerUseCaseImpl) GetCustomDirectQuote(ctx context.Context, tokenIn sd
 	return getBestSingleRouteQuote(ctx, tokenIn, routes, r.logger)
 }
 
+var ErrValidationFailed = fmt.Errorf("validation failed")
+
+// GetCustomDirectQuoteMultiPool implements mvc.RouterUsecase.
+func (r *routerUseCaseImpl) GetCustomDirectQuoteMultiPool(ctx context.Context, tokenIn sdk.Coin, tokenOutDenom []string, poolID []uint64) (domain.Quote, error) {
+	if len(poolID) == 0 {
+		return nil, fmt.Errorf("%w: at least one pool ID should be specified", ErrValidationFailed)
+	}
+
+	if len(tokenOutDenom) == 0 {
+		return nil, fmt.Errorf("%w: at least one token out denom should be specified", ErrValidationFailed)
+	}
+
+	// for each given pool we expect to have provided token out denom
+	if len(poolID) != len(tokenOutDenom) {
+		return nil, fmt.Errorf("%w: number of pool ID should match number of out denom", ErrValidationFailed)
+	}
+
+	var result quoteImpl
+	for i, v := range poolID {
+		tokenOutDenom := tokenOutDenom[i]
+
+		quote, err := r.GetCustomDirectQuote(ctx, tokenIn, tokenOutDenom, v)
+		if err != nil {
+			return nil, err
+		}
+
+		// fill the amount in token with the first asset of the pair
+		if i == 0 {
+			result.AmountIn = tokenIn
+		}
+
+		// fill the amount of the second asset of the pair
+		if i == len(poolID)-1 {
+			result.AmountOut = quote.GetAmountOut()
+		}
```

Comment:
To reduce cognitive overhead, can we update the `result.AmountOut` on every iteration?